pkg/check/porcupine/porcupine.go

```diff
@@ -0,0 +1,76 @@
+package porcupine
+
+import (
+	"log"
+
+	"github.com/pkg/errors"
+
+	"github.com/anishathalye/porcupine"
+	"github.com/siddontang/chaos/pkg/core"
+)
+
+// Checker is a linearizability checker powered by Porcupine.
+type Checker struct{}
+
+// Check checks the history of operations meets liearizability or not with model.
+// False means the history is not linearizable.
+func (Checker) Check(m core.Model, ops []core.Operation) (bool, error) {
+	pModel := porcupine.Model{
+		Init:  m.Init,
+		Step:  m.Step,
+		Equal: m.Equal,
+	}
+	events, err := ConvertOperationsToEvents(ops)
+	if err != nil {
+		return false, err
+	}
+	log.Printf("begin to verify %d events", len(events))
+	return porcupine.CheckEvents(pModel, events), nil
+}
+
+// Name is the name of porcupine checker
+func (Checker) Name() string {
+	return "porcupine_checker"
+}
+
+// ConvertOperationsToEvents converts core.Operations to porcupine.Event.
+func ConvertOperationsToEvents(ops []core.Operation) ([]porcupine.Event, error) {
+	if len(ops)%2 != 0 {
```

Comment:
should we check call and return

Reply:
The function requires a complete history of operation. Complete means there is a response(return) with respect to a given request(call).